syntax_and_semantics/cross-compilation.md

```diff
@@ -2,33 +2,60 @@
 
 Crystal supports a basic form of [cross compilation](http://en.wikipedia.org/wiki/Cross_compiler).
 
-In order to achieve this, the compiler executable provides two flags:
+## Preparation
+
+Cross-compilation will depend on `libcrystal` being present on the target system.  It can be built by following these steps:
+* Download the Crystal source code on the target machine.
+* Install the LLVM/compiler toolchain. (e.g. `sudo apt-get install clang` on Ubuntu)
+* Install the [required libraries](https://github.com/crystal-lang/crystal/wiki/All-required-libraries) for building Crystal.
+* Run `make deps` in the source directory.
+
+These steps will produce `libcrystal.a` within the `ext/src/` directory.
+
+
+## Compiling
+
+The compiler executable provides two flags:
 
 * `--cross-compile`: When given enables cross compilation mode
 * `--target`: the [LLVM Target Triple](http://llvm.org/docs/LangRef.html#target-triple) to use and set the default [compile-time flags](compile_time_flags.html) from
 
-To get the `--target` flags you can execute `llvm-config --host-target` using an installed LLVM 3.5. For example on a Linux it could say "x86_64-unknown-linux-gnu".
+The `--target` flags can be determined by executing `llvm-config --host-target` on the target system. (e.g. 'x86_64-unknown-linux-gnu')
```

Comment:
I suggest using `gcc -dumpmachine` instead, as `gcc` is far more common than `llvm-config`.

Reply:
Happy to switch this out, although the output seems less specific...

```bash
$ gcc -dumpmachine
arm-linux-gnueabihf
$ llvm-config --host-target
armv6-linux-gnueabihf
```

Admittedly I don't know how important this is.

Reply:
If it works as a target which you can cross-compile with, go for it.

Reply:
:+1: